discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -168,41 +168,57 @@ def cat(self, multihash):
             raise  # error is of type ipfshttpclient.exceptions.TimeoutError
 
     def multihash_is_directory(self, multihash):
+        # First attempt to cat multihash locally.
         try:
-            # attempt to cat single byte from CID to determine if dir or file
+            # If cat successful, multihash is not directory.
             self._api.cat(multihash, 0, 1)
             return False
         except Exception as e:  # pylint: disable=W0703
             if "this dag node is a directory" in str(e):
-                logger.warning(f'IPFSCLIENT | Found directory {multihash}')
+                logger.warning(f"IPFSCLIENT | Found directory {multihash}")
                 return True
 
-        # Attempt to retrieve from cnode gateway endpoints
+        # Attempt to retrieve from cnode gateway endpoints.
         gateway_endpoints = self._cnode_endpoints
         for address in gateway_endpoints:
-            gateway_query_address = "%s/ipfs/%s" % (address, multihash)
+            # First, query as dir.
+            gateway_query_address = "%s/ipfs/%s/150x150.jpg" % (address, multihash)
             r = None
             try:
-                logger.warning(f"IPFSCLIENT | Querying directory {gateway_query_address}")
+                logger.warning(f"IPFSCLIENT | Querying {gateway_query_address}")
                 r = requests.get(gateway_query_address, timeout=20)
```

Comment:
@hareeshnagaraj @SidSethi how do you feel about lowering this timeout to like 3 seconds? no point in having a timeout take > 6x indexing

Reply:
seems risky to reduce so much since the gateway fetch can occasionally take a while longer when it has to rehydrate. but the risk of reducing is low since this will raise exception and short circuit indexing task, and next task will retry. hareesh and i agreed on 10s as compromise

Reply:
fixed